src/server/engine_shard_set.cc

```diff
@@ -137,8 +141,8 @@ bool EngineShard::DoDefrag() {
   const float threshold = GetFlag(FLAGS_mem_defrag_page_utilization_threshold);
 
   auto& slice = db_slice();
-  DCHECK(slice.IsDbValid(kDefaultDbIndex));
-  auto [prime_table, expire_table] = slice.GetTables(kDefaultDbIndex);
+  DCHECK(slice.IsDbValid(defrag_state_.dbid));
```

Comment:
do we ever reduce number of our databases? i.e. flushall?

Reply:
We don't, see `DbSlice::FlushDb`

But I can change it a little nonetheless to make it safer